File: app/src/main/java/org/jqt3of5/android/dicebag/room/DiceInPlayDao.java
package org.jqt3of5.android.dicebag.room;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

/**
 * Created by Brittany on 4/28/2018.
 */
@Dao
public interface DiceInPlayDao
{
    @Query("SELECT * FROM DiceInPlay INNER JOIN DiceTemplates ON DiceInPlay.diceTemplateId = DiceTemplates.templateId")
    LiveData<List<FullDice>> getAllDiceInPlay();

    @Query("SELECT * FROM DiceInPlay INNER JOIN DiceTemplates ON DiceInPlay.diceTemplateId = DiceTemplates.templateId WHERE DiceInPlay.diceId = :diceId")
    FullDice getById(int diceId);

    @Insert
    void insertAll(List<DiceInPlayEntity> dice);

    @Delete
    void delete(DiceInPlayEntity dice);
}

File: app/src/main/java/org/jqt3of5/android/dicebag/DiceInPlayDao.kt
package org.jqt3of5.android.dicebag

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Delete
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query

/**
 * Created by Brittany on 4/28/2018.
 */
@Dao
interface DiceInPlayDao
{
    @Query("SELECT * FROM DiceInPlay NATURAL JOIN DiceTemplates")
    fun getAll() : LiveData<List<Dice>>

    @Query("SELECT * FROM DiceInPlay NATURAL JOIN DiceTemplates WHERE diceId = :diceId")
    fun getById(diceId : Int) : LiveData<Dice>

    @Insert
    fun insertAll(dice : Array<DiceInPlayEntity>)

    @Delete
    fun delete(dice : DiceInPlayEntity)
}